transaction.go

```diff
@@ -23,12 +23,121 @@ import "C"
 
 import (
 	"errors"
+	"fmt"
 	"runtime"
 	"runtime/cgo"
 	"strings"
+	"sync/atomic"
 	"unsafe"
 )
 
+// success indicates a successful function return.
+const success = C.PAM_SUCCESS
+
+// StatusError is the Type for PAM Return types
+type StatusError int
```

Comment:
This type could probably just be called `Error` instead of `StatusError`.

I think there is enough error-related code to move it all into a separate source file, maybe `errors.go`?